### ISSUE TITLE
Sync with installed Epic Games button does not work

### ISSUE THREAD

**Issue by iblowmymind** [Heroic-Games-Launcher/HeroicGamesLauncher#2943]
### Describe the bug

When clicking the button in settings, instead of asking me for a path (which is what I assume it does), it will just say "Error: Invalid Path" (see: screenshot 1)

I also noticed that the line "[cli] ERROR: Legendary is missing game metadata, please login (if not already) and use the "status" command to fetch necessary information to set-up syncing." gets logged whenever I try to do it, but I've been able to run the status command successfully (while also passing down XDG_CONFIG_HOME to ensure that changes get written to Heroic's legendary path) and run egl-sync on the command line to sync installed games, and I can confirm that after a relaunch my installed games do show up and I'm able to launch them.

There was another issue where I wasn't able to log in due to "UnicodeEncodeError: 'charmap' codec can't encode character '\u0130' in position 13: character maps to <undefined>" (Turkish issue :tm:), but that was an issue with legendary, and I was able to fix that by downloading the next point release and manually setting the legendary executable path.

### Add logs

```text
(11:17:58) INFO:    [Legendary]:        Running command: XDG_CONFIG_HOME=C:\Users\(redacted)\AppData\Roaming\heroic\legendaryConfig C:\dev\legendary.exe --version
(11:17:58) INFO:    [Legendary]:        Legendary location: C:\dev\legendary.exe
(11:17:58) INFO:    [Gog]:              GOGDL location: C:\Users\(redacted)\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\gogdl
(11:17:58) INFO:    [Connection]:       Connectivity: check-online
(11:17:58) INFO:    [Connection]:       Pinging external endpoints
(11:17:58) INFO:    [Backend]:          DRM module staus {
  "neifaoindggfcjicffkgpmnlppeffabd": {
    "name": "Google Widevine Windows CDM",
    "status": "new",
    "version": "1.0.2512.0"
  },
  "oimompecagnajdejgnnjijobebaeigek": {
    "name": "Widevine Content Decryption Module",
    "status": "new",
    "version": "4.10.2557.0"
  }
}
(11:17:58) WARNING: [Backend]:          Protocol already registered.
(11:17:58) INFO:    [Connection]:       Connectivity: online
(11:17:58) ERROR:   [Gog]:              Unable to syncQueued playtime, userData not present
(11:17:59) INFO:    [Gog]:              Running command: C:\Users\(redacted)\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\gogdl --auth-config-path C:\Users\(redacted)\AppData\Roaming\heroic\gog_store\auth.json --version
(11:17:59) INFO:    [Backend]:          Loading Screen Ready
(11:17:59) INFO:    [Frontend]:         Refreshing undefined Library
(11:17:59) WARNING: [Backend]:          refresh not implemented on Sideload Library Manager
(11:17:59) INFO:    [Legendary]:        Refreshing library...
(11:17:59) INFO:    [Legendary]:        Refreshing Epic Games...
(11:18:00) INFO:    [Legendary]:        Game list updated, got 156 games & DLCs
(11:18:00) INFO:    [Backend]:          Frontend Ready
(11:18:00) WARNING: [Backend]:          listUpdateableGames not implemented on Sideload Library Manager
(11:18:00) INFO:    [Gog]:              Found 0 game(s) to update
(11:18:00) INFO:    [Backend]:          Checking for new Heroic Updates
(11:18:00) INFO:    [Backend]:          Checking for current version changelog
(11:18:00) INFO:    [Legendary]:        Checking for game updates: XDG_CONFIG_HOME=C:\Users\(redacted)\AppData\Roaming\heroic\legendaryConfig C:\dev\legendary.exe list --third-party
(11:18:00) INFO:    [Legendary]:        Running command: XDG_CONFIG_HOME=C:\Users\(redacted)\AppData\Roaming\heroic\legendaryConfig C:\dev\legendary.exe list --third-party
(11:18:01) INFO:    [Nile]:             Running command: XDG_CONFIG_HOME=C:\Users\(redacted)\AppData\Roaming\heroic\nile_config C:\Users\(redacted)\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\nile --version
(11:18:03) WARNING: [Backend]:          Could not determine System Info
(11:18:03) INFO:    [Backend]:          VCRuntime is installed
(11:18:05) INFO:    [Backend]:          Starting the Download Queue
(11:18:05) DEBUG:   [Legendary]:        Update is available for 86ae9acccf8443e18fca6950e0181288: 2.26.0 != 2.27.0
(11:18:05) DEBUG:   [Legendary]:        Update is available for 963137e4c29d4c79a81323b8fab03a40: 3259 != 3365
(11:18:05) INFO:    [Legendary]:        Found 2 games to update
(11:18:07) INFO:    [Legendary]:        
(11:18:07) ERROR:   [Legendary]:        [cli] ERROR: Legendary is missing game metadata, please login (if not already) and use the "status" command to fetch necessary information to set-up syncing.

(11:18:44) INFO:    [Legendary]:        Running command: XDG_CONFIG_HOME=C:\Users\(redacted)\AppData\Roaming\heroic\legendaryConfig C:\dev\legendary.exe --version
(11:18:44) INFO:    [Gog]:              Running command: C:\Users\(redacted)\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\gogdl --auth-config-path C:\Users\(redacted)\AppData\Roaming\heroic\gog_store\auth.json --version
(11:18:44) INFO:    [Nile]:             Running command: XDG_CONFIG_HOME=C:\Users\(redacted)\AppData\Roaming\heroic\nile_config C:\Users\(redacted)\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\nile --version
(11:18:44) INFO:    [Legendary]:        Checking if EOS Overlay is enabled: XDG_CONFIG_HOME=C:\Users\(redacted)\AppData\Roaming\heroic\legendaryConfig C:\dev\legendary.exe eos-overlay info
(11:18:46) INFO:    [Legendary]:        installed.json updated, refreshing library
(11:18:46) INFO:    [Legendary]:        Abort command "XDG_CONFIG_HOME=C:\Users\(redacted)\AppData\Roaming\heroic\legendaryConfig C:\dev\legendary.exe eos-overlay info"
```

### Steps to reproduce

1. Go to Settings
2. Click on "Sync with installed Epic Games"
3. See error

### Expected behavior

A file dialog to show up and allow me to select my game directory, and games being synced.

### Screenshots

![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/28228415/2e132875-0071-45f8-bdc5-b850b041c676)

### Heroic Version

Latest Stable

### System Information

- OS: Windows 10 22H2 (19045.3208)
- CPU: Intel i5-9300H
- RAM: 16GB
- GPU: GTX 1050

### Additional information

After manually updating legendary, I was able to log out and log back in just fine, and I show up with my account name in the accounts menu.

**Comment by iblowmymind** [comment]
I was able to pinpoint the issue to legendary not being passed `XDG_CONFIG_DIR` when you run the EGL Sync through the settings. This means that it defaults to using `%userprofile%\.config\legendary` (the default config path of legendary), and failing (as there's nothing there). In any other operation from Heroic involving legendary, a `XDG_CONFIG_DIR` with a value of `%appdata%\heroic\legendaryConfig` is passed to make the legendary config redirect to Heroic's directory.

A temporary workaround to this is to create a symlink from `%userprofile%\.config\legendary` to `%appdata%\heroic\legendaryConfig\legendary` (reprecussions not fully tested, if there are any). And a fix would involve having our `XDG_CONFIG_DIR` passed on the "Sync with installed Epic Games" option as well, just like other options.

**Comment by stealthybox** [comment]
I am having a very similar issue with a new Windows 11 based install.
The only things I've done are log into an Epic Games account and attempt to enable `Sync with Installed Epic Games`.

Full log:
```
(14:12:41) INFO:    [Legendary]:        Running command: XDG_CONFIG_HOME=C:\Users\Leigh\AppData\Roaming\heroic\legendaryConfig C:\Users\Leigh\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary --version
(14:12:41) INFO:    [Legendary]:        Legendary location: C:\Users\Leigh\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary
(14:12:41) INFO:    [Gog]:              GOGDL location: C:\Users\Leigh\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\gogdl
(14:12:41) INFO:    [Connection]:       Connectivity: check-online
(14:12:41) INFO:    [Connection]:       Pinging external endpoints
(14:12:41) INFO:    [Backend]:          DRM module staus {
  "neifaoindggfcjicffkgpmnlppeffabd": {
    "name": "Google Widevine Windows CDM",
    "status": "new",
    "version": "1.0.2512.0"
  },
  "oimompecagnajdejgnnjijobebaeigek": {
    "name": "Widevine Content Decryption Module",
    "status": "new",
    "version": "4.10.2557.0"
  }
}
(14:12:41) WARNING: [Backend]:          Protocol already registered.
(14:12:41) INFO:    [Connection]:       Connectivity: online
(14:12:41) ERROR:   [Gog]:              Unable to syncQueued playtime, userData not present
(14:12:41) INFO:    [Gog]:              Running command: C:\Users\Leigh\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\gogdl --auth-config-path C:\Users\Leigh\AppData\Roaming\heroic\gog_store\auth.json --version
(14:12:41) INFO:    [Backend]:          VCRuntime is installed
(14:12:41) INFO:    [Backend]:          Loading Screen Ready
(14:12:41) INFO:    [Frontend]:         Refreshing undefined Library
(14:12:41) WARNING: [Backend]:          refresh not implemented on Sideload Library Manager
(14:12:41) INFO:    [Legendary]:        Refreshing library...
(14:12:41) INFO:    [Legendary]:        Refreshing Epic Games...
(14:12:41) INFO:    [Legendary]:        Game list updated, got 4 games & DLCs
(14:12:41) INFO:    [Backend]:          Frontend Ready
(14:12:41) WARNING: [Backend]:          listUpdateableGames not implemented on Sideload Library Manager
(14:12:41) INFO:    [Gog]:              Found 0 game(s) to update
(14:12:41) INFO:    [Backend]:          Checking for new Heroic Updates
(14:12:41) INFO:    [Backend]:          Checking for current version changelog
(14:12:41) INFO:    [Nile]:             Running command: XDG_CONFIG_HOME=C:\Users\Leigh\AppData\Roaming\heroic\nile_config C:\Users\Leigh\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\nile --version
(14:12:42) INFO:    [Legendary]:        Checking for game updates: XDG_CONFIG_HOME=C:\Users\Leigh\AppData\Roaming\heroic\legendaryConfig C:\Users\Leigh\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary list --third-party
(14:12:42) INFO:    [Legendary]:        Running command: XDG_CONFIG_HOME=C:\Users\Leigh\AppData\Roaming\heroic\legendaryConfig C:\Users\Leigh\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary list --third-party
(14:12:43) WARNING: [Legendary]:        Failed to parse games from C:\Users\Leigh\AppData\Roaming\heroic\legendaryConfig\legendary\installed.json with: Error: ENOENT: no such file or directory, open 'C:\Users\Leigh\AppData\Roaming\heroic\legendaryConfig\legendary\installed.json'
    at Object.openSync (node:fs:600:3)
    at Object.func [as openSync] (node:electron/js2c/asar_bundle:2:1861)
    at Object.readFileSync (node:fs:468:35)
    at t.readFileSync (node:electron/js2c/asar_bundle:2:9204)
    at Module.listUpdateableGames (C:\Users\Leigh\AppData\Local\Programs\heroic\resources\app.asar\build\electron\main.f23c4159.js:102:4644)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async C:\Users\Leigh\AppData\Local\Programs\heroic\resources\app.asar\build\electron\main.f23c4159.js:191:3990
    at async node:electron/js2c/browser_init:2:100305
(14:12:43) INFO:    [Legendary]:        Found 0 games to update
(14:12:43) DEBUG:   [Backend]:          Unable to load Steam Libraries, libraryfolders.vdf not found
(14:12:44) INFO:    [Backend]:          

Heroic Version: 2.9.1 Boa Hancock
Legendary Version:  0.20.32 Dark Energy (hotfix #6)
GOGdl Version: 0.7.3
Nile Version: 1.0.0 Jonathan Joestar

Electron Version: 24.4.1
Chrome Version: 112.0.5615.204
NodeJS Version: 18.14.0

OS: Microsoft Windows 11 Home KERNEL: 10.0.22621 ARCH: x64
CPU: AMD Ryzen Z1 Extreme @3.3 
RAM: Total: 11.7 GiB Available: 4.43 GiB
GRAPHICS: GPU0: AMD Radeon Graphics VRAM: 4096MB 


(14:12:46) INFO:    [Backend]:          Starting the Download Queue
(14:13:22) DEBUG:   [Backend]:          Unable to load Steam Libraries, libraryfolders.vdf not found
(14:13:30) INFO:    [Legendary]:        
(14:13:30) ERROR:   [Legendary]:        [cli] ERROR: Legendary is missing game metadata, please login (if not already) and use the "status" command to fetch necessary information to set-up syncing.

```

**Comment by stealthybox** [comment]
I tried @iblowmymind's workaround of symlinking
`%userprofile%\.config\legendary` -> `%appdata%\heroic\legendaryConfig\legendary`
and it worked.
1. search cmd.exe in start menu
2. right click -> "Run as administrator"
3. run the following two commands:
```
move %userprofile%\.config\legendary %userprofile%\.config\legendary.bak
mklink /D %userprofile%\.config\legendary %appdata%\heroic\legendaryConfig\legendary
```

"Sync with Installed Epic Games" in the Heroic settings should work now.

**Comment by arielj** [comment]
A fix for this is already merged in the main branch, it should be solved in the next 2.9.2 release

**Comment by arielj** [comment]
This should have been fixed in 2.9.2
